packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1595,6 +1639,40 @@ class EditableText extends StatefulWidget {
   /// {@macro flutter.services.TextInputConfiguration.enableIMEPersonalizedLearning}
   final bool enableIMEPersonalizedLearning;
 
+  /// {@template flutter.widgets.editableText.contentInsertionMimeTypes}
+  /// Used when a user inserts image-based content through the device keyboard,
+  /// currently only used on Android.
+  ///
+  /// The passed list of strings will determine which MIME types are allowed to
+  /// be inserted via the device keyboard.
+  ///
+  /// The default mime types are given by [kDefaultContentInsertionMimeTypes].
+  /// These are all the mime types that are able to be handled and inserted
+  /// from keyboards.
+  ///
+  /// This field is governed by three different cases:
+  /// - If onContentInserted is not provided, an empty list will be passed to
+  /// the engine to indicate that the text field does not support handling
+  /// content insertion.
+  /// - If onContentInserted is provided but this field is not,
+  /// [kDefaultContentInsertionMimeTypes] will be passed to the engine instead.
+  /// - If both are provided, this field will be passed as-is to the engine.
+  ///
+  /// Note that if this field is provided, onContentInserted must also be provided.
```

Comment:
We should make sure this statement is still true after we decide on the defaults discussed above.

Reply:
Reworked this statement as per changes made.